benchmarktests/target_auth_approle.go

```diff
@@ -85,7 +80,9 @@ type SecretIDConfig struct {
 // test configuration in Vault. Any default configuration definitions for required
 // parameters will be set here.
 func (a *ApproleAuth) ParseConfig(body hcl.Body) error {
-	a.config = &ApproleTestConfig{
+	testConfig := &struct {
+		Config *ApproleAuthTestConfig `hcl:"config,block"`
+	}{
```

Comment:
I get why were doing this...The top level struct does seem a bit redundant. But I think this is more confusing :/ ... I could be just used to the other way, and will defer to what others think!

Reply:
The top level struct only really exists to allow for the decoding to work as expected here, and we're generally only concerned with the inner struct anyway. While it is more verbose, having this outer struct made naming a bit more difficult considering we really didn't do anything with it after handling the HCL decoding.